pkg/oc/cli/cmd/importer/appjson.go

```diff
@@ -56,67 +59,68 @@ var (
 )
 
 type AppJSONOptions struct {
-	Action configcmd.BulkAction
+	PrintFlags *genericclioptions.PrintFlags
 
-	In        io.Reader
-	Filenames []string
+	Printer printers.ResourcePrinter
 
-	BaseImage  string
-	Generator  string
-	AsTemplate string
+	bulkAction configcmd.BulkAction
+
+	BaseImage        string
+	Generator        string
+	AsTemplate       string
+	OutputVersionStr string
 
-	PrintObject    func(runtime.Object) error
 	OutputVersions []schema.GroupVersion
 
 	Namespace string
 	Client    templateclient.TemplateInterface
+
+	genericclioptions.IOStreams
+	resource.FilenameOptions
 }
 
-// NewCmdAppJSON imports an app.json file (schema described here: https://devcenter.heroku.com/articles/app-json-schema)
-// as a template.
-func NewCmdAppJSON(fullName string, f kcmdutil.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
-	options := &AppJSONOptions{
-		Action: configcmd.BulkAction{
-			Out:    out,
-			ErrOut: errout,
+func NewAppJSONOptions(streams genericclioptions.IOStreams) *AppJSONOptions {
+	return &AppJSONOptions{
+		bulkAction: configcmd.BulkAction{
+			Out:    streams.Out,
+			ErrOut: streams.ErrOut,
 		},
-		In:        in,
-		Generator: AppJSONV1GeneratorName,
+		IOStreams:  streams,
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(ocscheme.PrintingInternalScheme),
```

Comment:
"created"?

Reply:
Added this change in https://github.com/openshift/origin/pull/20269